src/osd/PG.cc

```diff
@@ -4764,6 +4764,23 @@ void PG::scrub_compare_maps()
       maps[*i] = &scrubber.received_maps[*i];
     }
 
+    set<hobject_t> seen;
+    for (auto map : maps) {
+      for (auto object : map.second->objects) {
+        if (object.second.large_omap_object_found
+            && seen.find(object.second.large_omap_object) == seen.end()) {
+          scrubber.large_omap_objects++;
+          osd->clog->warn() << "Large omap object found. Object: "
+                            << object.second.large_omap_object
+                            << " Key count: "
+                            << object.second.large_omap_object_key_count
+                            << " Size (bytes): "
+                            << object.second.large_omap_object_value_size;
+          seen.insert(object.second.large_omap_object);
+        }
+      }
+    }
+
```

Comment:
This code is inside _if (acting.size() > 1) {_, so it won't check if there is only 1 shard.  I suggest you write a new function modeled after the start of be_compare_scubmaps().  Call it before the check acting.size() check.  You'll have to move the creation of _maps_ earlier.


   map<pg_shard_t, ScrubMap *> maps
...
   maps[pg_whoami] = &scrubber.primary_scrubmap;

    for (set<pg_shard_t>::iterator i = actingbackfill.begin();
         i != actingbackfill.end();
         ++i) {
      if (*i == pg_whoami) continue;
      dout(2) << __func__ << " replica " << *i << " has "
              << scrubber.received_maps[*i].objects.size()
              << " items" << dendl;
      maps[*i] = &scrubber.received_maps[*i];
    }


Doing this will eliminate the need for large_omap_object because each ScrubMap::object is already in the context of a specific hobject_t and pg_shard_t.